CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+> modify the terms of any separate license agreement you may have executed
+> with Licensor regarding such Contributions.
+
+## Commits
+
+Please follow established best practices for your commits.
+Do only one thing per commit, and write a descriptive commit message.
+[Chris Beams' *seven rules of a great Git commit
+message*](https://chris.beams.io/posts/git-commit/#seven-rules)
+provides a good set of guidelines:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain *what* and *why* vs. *how*
+
+## Pull Requests
+
+Please follow established best practices for your pull requests.
+Give a short description of *what* your patch does in the title,
+and include a short description of *why* the patch is needed in the comment.
+A complex title or summary is a good indication that your pull request
+is doing too many things, and should be split into multiple patches.
+
+See also [*10 tips for better Pull Requests* by Mark
+Seemann](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/).
+
+When you create a pull request, please assign one or more core team members as 
+the reviewers. You should also assign all other contributors who may be affected
+by your changes.
+
+Once you create a pull request, GitHub will automatically start a discussion 
+concerning your request. You should monitor the discussion and address the suggestions,
+concerns and critiques from other collaborators.
+
+### Merging Pull Requests
+
+Only the Core team members have the merge privileges for pull requests in the master branch, but all collaborators
+are welcome to review and discuss a particular pull request. Here's a nice set of 
+guidelines to follow when reviewing and discussing contributor's code: 
+https://github.com/thoughtbot/guides/tree/master/code-review
+
+Once a consensus has been reached among the contributors, a core member can manually perform the merge. 
+
+Core members should **never** merge their own pull requests, unless in a high emergency situation (for example, a Core member
+has mistakenly pushed some proprietary or security-sensitive data into the master and needs to immediately revert the change).
+
+## Quality Control for Contributed Code
+
+### Regression Tests
+
+The main regression tests for HClib are located in `test/c` and `test/cpp`.
+The `test_all.sh` scripts in each of those directories will automatically
+build and run all tests in the respective test suite.
+
+All new commits should pass all of the regression tests.
+When you open a pull request, Travis-CI will automatically run
+the full regression suite against your cumulative change
+(not against individual commits).
+Please run the tests locally against each commit to ensure
```

Comment:
Should we ask for each pull request to be squashed to a single commit, in the spirit of making small, incremental, separable changes? Which would also make this advice unnecessary.